broker/endpoints/Patient.js

```diff
@@ -2,30 +2,99 @@ const axios = require("../services/axiosInstance.js");
 const PatientIdService = require("../services/PatientIdService");
 const Patient = require("../models/Patient");
 const RelatedPerson = require("../models/RelatedPerson");
+const {body, validationResult} = require('express-validator');
 const Appointment = require("../models/Appointment");
 
 const prepend = "x";
 
 exports.prepend = prepend;
 
 exports.read = (req, res) => {
-  axios
-    .get(`${req.url}`)
+  
+  if(!req.params.qrCode) {
+    return res.status(400).json({
+      error: "No QR Code provided for patient lookup"
+    })
+  }
+
+  PatientIdService.getPatientIdForQrCode(req.params.qrCode)
+  .then((patientIdRecord) => {
+    const patientId = patientIdRecord.patient_id;
+    if(!patientId) {
+      throw {
+        status: 404,
+        message: "No patient found with that QR Code"
+      }
+    }
+    return axios.get(`${process.env.FHIR_URL_BASE}/Patient/${prepend}${patientId}`);
+  })
+  .then((patientPayload) => {
+    const patientRecord = Patient.toModel(patientPayload.data);
+    return res.json(patientRecord);
+  })
+  .catch((err) => {
+    res.status(err.status || 400).json({
+      error: err.response ? err.response.data : err.message,
+    });
+  });
+};
+
+exports.search = [
+
+  // Demonstrates standard usage of express-validator
+  // Note that exports.search is now an array of functions, rather than a single function
+  // Docs: https://express-validator.github.io/docs/index.html
+  body('firstName', "Please enter the patient's first name").isLength({ min: 1}).trim().escape(),
+  body('lastName', "Please enter the patient's last name").isLength({ min: 1 }).trim().escape(),
+  body('birthDate', "Please enter a valid patient birth date").isISO8601({ strict: true}).toDate(),
+  body('postalCode', "Invalid postal code entered").optional({checkFalsy: true}).trim().isNumeric().isLength({ min: 5, max: 5}),
```

Comment:
Good catch, I updated it!  That got added since my original dev  :P   Side note, I mentioned to Dana that she had made ZIP required, and she asked that I remove it for her.  So I slid that change in today as well with her blessing.